### PR TITLE
Associations to Development branch

### PR DESCRIPTION
**Pull request by CalyCherkaoui**
In this feature branch, we created our models: User and Event, and associated them with a polymorphic table "attendence".
We also created the basic views and controllers corresponding to our models.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      <li> Title: <%= event.title %> Location: <%= event.location %> Date: <%= event.event_date %> </li>
+    <% end %>
+</ul>
+
+
+<h2>List of Previous events </h2>
+<% @previous_events = @user.attended_events.where('event_date < ?', Time.now).to_a %>
+<% @upcomming_events = @user.attended_events.where('event_date > ?', Time.now).to_a %>
+<ul>
+  <% @previous_events.each do |event| %>
```
Comment (wathigo) [inline app/views/users/show.html.erb]: **REQUIRED**
There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.
Anything related to data processing should be placed in models passed to the views via the controllers.
**Resources.**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+<p>
+  <strong>Username:</strong>
+  <%= @user.username %>
+</p>
+
+<h2> List of <%= @user.username %> creates events: </h2>
+<ul>
+  <% @events.each do |eve| %>
+    <li> Title: <%= eve.title %> Location: <%= eve.location %> </li>
+  <% end %>
```
Comment (wathigo) [inline app/views/users/show.html.erb]: **REQUIRED**
* [x] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)).
Let's separate user created events in past and upcoming.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  has_many :events, foreign_key: :creator_id, class_name: 'Event'
```
Comment (wathigo) [inline app/models/user.rb]: There is no need of adding `class_name: 'Event'`. Rails is able to process this.

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required.
Great work guys :clap:
Let's make some changes before we move on.

See the comments for more details.

**OPTIONAL**
* We can prevent a user from attending an event if it's in the past.
* Let's add some styles to the app to improve UI.

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo.**

**After making the changes, submit another request for review.**

File: app/views/users/show.html.erb (hunk 1 above)
Comment (CalyCherkaoui) [reply app/views/users/show.html.erb]: Thank you for the review. Issue solved.

File: app/models/user.rb (hunk 3 above)
Comment (CalyCherkaoui) [reply app/models/user.rb]: Thanks. Issue solved.

File: app/views/users/show.html.erb (hunk 2 above)
Comment (CalyCherkaoui) [reply app/views/users/show.html.erb]: Thank you! Issue solved.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  # GET /users/1
+  # GET /users/1.json
+  def show
+    @user = User.find(params[:id])
+    @events = Event.where(creator_id: @user.id).all
+    @past_events = @events.past
+    @future_events = @events.upcoming
+    @attended_events = @user.attended_events
+    @previous_events = @attended_events.where('event_date < ?', Time.now).to_a
```
Comment (wathigo) [inline app/controllers/users_controller.rb]: **REQUIRED**
* Lists events previoulsy attendented by user **using event model scope**, i.e. ` @previous_events = current_user.attended_events.past` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    @upcomming_events = @attended_events.where('event_date > ?', Time.now).to_a
```
Comment (wathigo) [inline app/controllers/users_controller.rb]: **REQUIRED**
* Lists upcoming events previously attended by user **using event model scope**, i.e. `@upcomming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required.
Great work implementing the suggested changes.
There is one more problem we need to address.
See the comments :point_down: 

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo.**

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (asiimwedismas) [reply app/controllers/users_controller.rb]: Fixed

**Comment by Dandush03** [comment]
# Status: 🚀 Approved 🚀 

### Hey! Great job so far! 💪 
### Keep it up in your next project
# Happy Coding 🤓 💻

**Pull request merged**